keps/sig-node/3857-rro-mounts/README.md

```diff
@@ -0,0 +1,1275 @@
+The pod manifest has to be recreated to get a consistent state in this case.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Works.
+Just same as a fresh roll-out, as long as the user has recreated the pod manifests.
+(See "Can the feature be disabled once ..." section above)
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+Unit tests will run with and without the feature gate.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+A rollout may fail when at least one of the following components are too old:
+
+| Component      | `readOnlyRecursive` value that will cause an error |
+|----------------|----------------------------------------------------|
+| kube-apiserver | any value                                          |
+| kubelet        | any value                                          |
+| CRI runtime    | `Enabled`                                          |
+| OCI runtime    | `Enabled`                                          |
+| kernel         | `Enabled`                                          |
+
+For example, an error will be returned like this if kube-apiserver is too old:
+```console
+$ kubectl apply -f rro.yaml
+Error from server (BadRequest): error when creating "rro.yaml": Pod in version "v1" cannot be handled as a Pod:
+strict decoding error: unknown field "spec.containers[0].volumeMounts[0].recursiveReadOnly"
+```
+
+No impact on already running workloads.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+Look for an event saying indicating RRO is not supported by the runtime.
```

Comment:
events are not easy for a cluster admin to observe. metrics are much better. So, a few things I can imagine we might need to look for:
- crashing kubelet (especially when the field is set)
- pod failures / errors from the runtime
- pod rejections

Imagine you enable this on a cluster with 100 different users. How do you know if it's causing a problem? If it is causing a problem for some users, but other users are relying on it, can you roll it back?

Now imagine you have 10,000 clusters. You enable this on 100 clusters. You want to know if you should enable it on more clusters. How can you tell if users are using it? How can you tell if it is working?

That's what we're trying to get to with some of these questions.

Note these are non-blocking for alpha.